### PR TITLE
Add Gson.

### PR DESCRIPTION
**Pull request by ThatRedox**
* Add Gson and licensing info.
* Switch resource pack loading to use Gson in lenient mode.

### CODE REVIEW

**Comment by leMaik** [comment]
Can we remove llbit's json library then?

**Comment by ThatRedox** [comment]
Not yet, but once we transition everything to use Gson, yes.

**Comment by leMaik** [comment]
Should we continue the switch in a follow-up PR, eg. for the launcher?

File: README.md [hunk 1]
```diff
@@ -1,224 +1,227 @@
...
+  The library is covered by the Apache License, version 2.0.
+See the file `licenses/Apache-2.0.txt` for the full license text.
+See the file `licenses/commons-math.txt` for the copyright notices.
+* **FastUtil by Sebastiano Vigna**.
+  FastUtil is covered by Apache License, version 2.0. 
+See the file `licenses/Apache-2.0.txt` for the full license text.
+See the file `licenses/fast-util.txt` for the copyright notice.
+* **Gson by Google**
+  The library is convered by the Apache License, version 2.0.
+See the file `licenses/Apache-2.0.txt` for the full license text.
```
Comment (leMaik) [inline README.md:201]: We should add a `licenses/gson.txt` file that contains the copyright notice, ie. this block of text: https://github.com/google/gson#license

**Review by leMaik — APPROVED** [review]

**Pull request merged**